src/sbvr-api/sbvr-utils.ts

```diff
@@ -1567,22 +1682,29 @@ const prepareResponse = async (
 	result: any,
 	tx: Db.Tx,
 ): Promise<Response> => {
+	let response: Response;
 	switch (request.method) {
 		case 'GET':
-			return await respondGet(req, request, result, tx);
+			response = await respondGet(req, request, result, tx);
+			break;
 		case 'POST':
-			return await respondPost(req, request, result, tx);
+			response = await respondPost(req, request, result, tx);
+			break;
 		case 'PUT':
 		case 'PATCH':
 		case 'MERGE':
-			return await respondPut(req, request, result, tx);
+			response = await respondPut(req, request, result, tx);
+			break;
 		case 'DELETE':
-			return await respondDelete(req, request, result, tx);
+			response = await respondDelete(req, request, result, tx);
+			break;
 		case 'OPTIONS':
-			return await respondOptions(req, request, result, tx);
+			response = await respondOptions(req, request, result, tx);
+			break;
 		default:
 			throw new MethodNotAllowedError();
 	}
+	return { ...response, id: request.batchRequestId };
```

Comment:
As we don't want to return the id: as undefined when we are in the single request processing, it should better not return the id key at all. 
```suggestion
	return { ...response, id: request.batchRequestId ?? null }; // or
    return { ...response, ... ( request.batchRequestId ? {id: request.batchRequestId} :  null) }; or
    
    // Potentially the preferd one, as it's not needing the `...` object expand operator that could be heavy depending on the return object size.
    if (request.batchRequestId != null ) {
        response['id'] = request.batchRequestId
    }
    return response;
```